### PR TITLE
Try to deal with issue #43 r

### PR DESCRIPTION
**Pull request by roachspray**
Try to deal with issue #43 related to resource strings. Maybe not the correct route given to_string(), but let's see.

### CODE REVIEW

**Comment by Ghabry** [comment]
Imo this is good enough that way because. But using codecvt is a problem (as travis shows) because this wasn't added before gcc5 :/.

Try to google some utf-16 to utf-8 code ^^

**Comment by roachspray** [comment]
@Ghabry Oh cripes... yea, that was a bit naive of me (and a bit too late in the night for me to be submitting PRs :-)). Thanks.

File: pe-parser-library/src/parse.cpp [hunk 1]
```diff
@@ -193,7 +193,7 @@ void IterRsrc(parsed_pe *pe, iterRsrc cb, void *cbd) {
   return;
 }
 
-bool parse_resource_id(bounded_buffer *data, std::uint32_t id, string &result) {
+bool parse_resource_id(bounded_buffer *data, std::uint32_t id, u16string &result) {
```
Comment (pgoodman) [inline pe-parser-library/src/parse.cpp]: `std::u16string`.

File: dump-pe/main.cpp [hunk 2]
```diff
@@ -227,17 +227,20 @@ int printSymbols(void *N,
 
 int printRsrc(void *N, resource r) {
   static_cast<void>(N);
+  std::basic_string<char> name(r.name_str.begin(), r.name_str.end());
```
Comment (jacwil) [inline dump-pe/main.cpp]: In Windows, this doesn't convert from UTF-16 to UTF-8. It actually just breaks the each `char16_t` into 2 `char`s. I'd expect this is the same behavior on Linux.

**Comment by jacwil** [comment]
This isn't working on Windows. The conversion from utf-16 to utf-8 isn't happening.

**Comment by jacwil** [comment]
Can't we update the version of gcc in Travis so it supports codecvt? This seems to be the best platform-independent way to do Unicode conversion...

**Comment by woodruffw** [comment]
I'm 👍 for `codecvt`, although I think we need just `codecvt_utf8<int16_t>` instead of `codecvt_utf8_utf16` (confusingly, the former does UCS-2 when passed a `int16_t` or `char16_t` and I'm pretty sure that's what Windows uses, not "real" UTF-16).

**Pull request closed without merging**